### PR TITLE
Add guards to cluster deletion from cli

### PR DESCRIPTION
**Pull request by luca3rd**
## What does this PR do?

Adds guards to cluster deletion.
- If cluster has running apps -> throw an error
- If cluster has stopped apps -> confirm w/ user that apps and logs will be deleted

### Deleting a cluster with running apps
Screen Shot 2022-12-14 at 11 39 48 AM

### Deleting a cluster with non-running apps
Screen Shot 2022-12-14 at 11 42 43 AM

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Review by nicolai86 — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]

File: src/lightning_app/cli/cmd_clusters.py [hunk 1]
```diff
@@ -198,6 +203,33 @@ def delete(self, cluster_id: str, force: bool = False, do_async: bool = False) -
...
+                )
+
+            if _list_apps(self.api_client, cluster_id=cluster_id, phase_not_in=[V1LightningappInstanceState.DELETED]):
+                click.echo(
+                    dedent(
+                        """\
+                        This cluster has non-running apps.
+                        Deleting this cluster will delete those apps and their logs.
+
+                        App data will still be available in the S3 bucket for the cluster.
```
Comment (tchaton) [inline src/lightning_app/cli/cmd_clusters.py:227]:
```suggestion
                        App artifacts aren't deleted and still accessible in the S3 bucket for the cluster.
```

File: src/lightning_app/cli/cmd_clusters.py [hunk 2]
```diff
@@ -198,6 +203,33 @@ def delete(self, cluster_id: str, force: bool = False, do_async: bool = False) -
...
+                        """\
+                        To delete the cluster, you must first delete the apps running on it.
+                        Use the following commands to delete the apps, then delete the cluster again:
+
+                        """
+                    )
+                    + "\n".join([f"\tlightning delete app {app.name} --cluster-id {cluster_id}" for app in apps])
+                )
+
+            if _list_apps(self.api_client, cluster_id=cluster_id, phase_not_in=[V1LightningappInstanceState.DELETED]):
```
Comment (tchaton) [inline src/lightning_app/cli/cmd_clusters.py:220]: Question: Won't DELETED instance not be listed ?

File: src/lightning_app/cli/cmd_clusters.py [hunk 3]
```diff
@@ -198,6 +203,33 @@ def delete(self, cluster_id: str, force: bool = False, do_async: bool = False) -
...
+                click.echo(
+                    dedent(
+                        """\
+                        This cluster has non-running apps.
```
Comment (tchaton) [inline src/lightning_app/cli/cmd_clusters.py:224]: This isn't super clear for the end user.

**Review by tchaton — APPROVED** [review]
LGTM !

File: src/lightning_app/cli/cmd_clusters.py (hunk 3 above)
Comment (luca3rd) [reply src/lightning_app/cli/cmd_clusters.py:224]: I'll change this to 'stopped apps'

File: src/lightning_app/cli/cmd_clusters.py (hunk 2 above)
Comment (luca3rd) [reply src/lightning_app/cli/cmd_clusters.py:220]: If `phase_not_in` and `phase_in` are not provided the default behaviour is to return all instances that are not deleted.

I could change the query to use this default behaviour, but it doesn't hurt to be explicit either.

**Pull request merged**